include/internal/tlsgroups.h

```diff
@@ -56,5 +56,8 @@
 # define OSSL_TLS_GROUP_ID_ffdhe4096        0x0102
 # define OSSL_TLS_GROUP_ID_ffdhe6144        0x0103
 # define OSSL_TLS_GROUP_ID_ffdhe8192        0x0104
+/* TODO: Update to 262 as per https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-8 */
+/* Not done yet to not break interop testing with OQS test server; change when that gets updated to IANA */
+# define OSSL_TLS_GROUP_ID_mlkem768         0x0768
```

Comment:
`0x11EB` and `0x11EC` are supported by the tlslite-ng and tlsfuzzer, just need to update the submodules, add kyber-py library and you can use https://github.com/tlsfuzzer/tlsfuzzer/blob/master/scripts/test-tls13-mlkem.py for testing

Reply:
> This should be fixed before merging 

It will be. oqs-provider has the required update, it's just not at the interop server.

> 0x11EB and 0x11EC are supported by the tlslite-ng and tlsfuzzer

Thanks for the pointer, @tomato42 . Those are the hybrid KEMs code points, though, while this PR is the plain ML-KEM. Hybrid's next.

Reply:
aah, I didn't see https://datatracker.ietf.org/doc/html/draft-connolly-tls-mlkem-key-agreement-03 
but then it hasn't been discussed on the TLS WG ML?

I haven't seen much enthusiasm for plain ML-KEM on the mailing list... I think it would be a mistake to include support for them in OpenSSL...

Reply:
:woman_facepalming:  actually the id 262 for MLKEM goes against RFC 7919, it should have not been registed by IANA

Reply:
> I think it would be a mistake to include support for them in OpenSSL.

Whether or not this is going to be included can be decided much later in the game (basically a one-line change to support it or not). Right now this is to check ML-KEM768 crypto works in OpenSSL (and as `oqsprovider` also supports it, it allows simple interop testing until more stringent KAT etc. tests follow).

> actually the id 262 for MLKEM goes against RFC 7919, it should have not been registed by IANA

Are you going to tell IANA?

Reply:
> Are you going to tell IANA?

No need, it seems, @tomato42 : Others have observed this too: https://github.com/open-quantum-safe/oqs-provider/issues/550#issuecomment-2452122731. So for now, will leave as-is.

Reply:
It can be merged to the feature branch in this temporary state if `TODO` is changed to `TODO(ML-KEM)`. We will not merge the feature branch to master with any outstanding `TODO(ML-KEM)` comments.

(A nit: also change this to a proper multi-line comment)

Reply:
Will change TODO's suitably. Many other style changes already done. One question, though: I did a little change to "include/internal/constant_time.h" -- and that fails the style check in _many_ more places than the ones I changed. Can I safely assume I can leave that file alone or should I change all that code to make `util/check-formal.pl` happy?

Reply:
Do style changes only for the lines that you are changing anyway. Unrelated style changes would just create unnecessary conflicts on rebases.

Reply:
> > Are you going to tell IANA?
> 
> No need, it seems, @tomato42 : Others have observed this too: [open-quantum-safe/oqs-provider#550 (comment)](https://github.com/open-quantum-safe/oqs-provider/issues/550#issuecomment-2452122731). So for now, will leave as-is.

not others... I'm Alicja :grin:

Reply:
> not others... I'm Alicja 😁

Oops -- I should have followed the link above one more click :-/ I now kind of appreciate better the OpenSSL position to only include stuff that's fully standardized.... Thanks for getting this rectified before it's causing too much work and confusion!

> I'd really like to see ML-KEM-1024 too, not just ML-KEM-768... any plans on that?

Yes: I just wanted to see how well the boringssl code ports (and "behaves") within OpenSSL before bringing that over too.